### ISSUE TITLE
Seg faults when running on slurm

### ISSUE THREAD

**Issue by Chris-Pedersen** [Chris-Pedersen/torch_qg#20]
Getting this quite commonly when running on slurm, but not through ood. Often can be a memory issue - but this latest one happened right at script start: 

```
WARNING: Could not find any nv files on this host!
/ext3/miniconda3/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3483.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/bin/bash: line 1: 1999515 Segmentation fault      (core dumped) python3 /home/cp3759/Projects/torch_qg/scripts/gen_L_sims.py
```

Then another error for the smagorinsky sims:

```
WARNING: Could not find any nv files on this host!
/ext3/miniconda3/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3483.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
corrupted double-linked list
/bin/bash: line 1: 2532150 Aborted                 (core dumped) python3 /home/cp3759/Projects/torch_qg/scripts/gen_smag_sims.py
```

**Comment by Chris-Pedersen** [comment]
From https://stackoverflow.com/questions/14897157/what-does-corrupted-double-linked-list-mean looks like both are related to memory issues. Given I've never had problems with identical syntax on ood, could be an issue with memory allocation on cpu. Try reproduce on interactive node, and contact support if it persists

**Comment by Chris-Pedersen** [comment]
Identical script working fine on interactive node. Memory requirements on my submitted slurm script were 120GB, memory on my current interactive node is 64GB and it's already gone further than the submitted script.

**Comment by Chris-Pedersen** [comment]
Crashing on interactive too:

```
[cp3759@log-2 scripts]$ srun -c4 -t2:00:00 --mem=64000 --pty /bin/bash
srun: job 40457741 queued and waiting for resources
srun: job 40457741 has been allocated resources
[cp3759@cm023 scripts]$ ls
gen_anim.py  gen_L_sims.py  gen_smag_sims.py    slurm_40448373.out  torchqg.mp4
gen_jet.py   gen_sims.py    slurm_40447940.out  submit_cpu.sh       torchqg_smag.mp4
[cp3759@cm023 scripts]$ sing
Singularity> cond
Singularity> python3 gen_smag_sims.py 
/ext3/miniconda3/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3483.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
Saving as /scratch/cp3759/pyqg_data/sims/torchqg_tests/arakawa_s0.01_48.nc
Saving as /scratch/cp3759/pyqg_data/sims/torchqg_tests/arakawa_s0.01_64.nc
Saving as /scratch/cp3759/pyqg_data/sims/torchqg_tests/arakawa_s0.01_128.nc
corrupted size vs. prev_size
```

**Comment by Chris-Pedersen** [comment]
Also related to issues in #23 , where I'm getting

```
>>> import torch_qg.model as torch_model
>>> import torch_qg.parameterizations as torch_param
>>> model=torch_model.ArakawaModel(nx=256,dt=3600,parameterization=torch_param.Smagorinsky())
/ext3/miniconda3/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3483.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
>>> ds=model.run_sim(int(8e4))
  0%|                                                                                                                                                                                                                                                                                                         | 0/80000 [00:00<?, ?it/s]free(): corrupted unsorted chunks
Aborted (core dumped)
```

**Comment by Chris-Pedersen** [comment]
Memory issues are coming from the Smagorinsky parameterisation - can run with gridsize of 3k on ood fine without smag. But even 128 goes oom when including smag. Some leak somewhere.

**Comment by Chris-Pedersen** [comment]
Appears to be fixed in #23 , memory issues appeared to be being caused by the fft input/output dims not being set (although suprised torch fft didn't flag this, if both the output was the wrong quantity, and it led to memory issues).
